packages/vite/src/node/utils.ts

```diff
@@ -775,8 +775,8 @@ function gracefulRename(
     ) {
       setTimeout(function () {
         fs.stat(to, function (stater, st) {
-          if (stater && stater.code === 'ENOENT') gracefulRename(from, to, CB)
-          else cb(er)
+          if (stater && stater.code === 'ENOENT') fs.rename(from, to, CB)
+          else CB(er)
```

Comment:
This seems to be a bug on node-graceful-fs. If the code isn't `'ENOENT'`, it means the directory still exists and we should keep retrying. So we need to call `CB`. Calling `cb` ends the recursion returning the original error.

Reply:
Great catch. This makes sense to me too after catching the flow.